src/main/java/com/minecolonies/entity/ai/citizen/fisherman/EntityAIWorkFisherman.java

```diff
@@ -210,8 +200,17 @@ private AIState startWorkingAtOwnBuilding()
      */
     private AIState prepareForFishing()
     {
-        if (checkOrRequestItems(new ItemStack(Items.FISHING_ROD)))
+        if (checkOrRequestItems(new ItemStack(Items.FISHING_ROD)) && worker != null)
```

Comment:
The worker != null has to be checked somewhere else.
With the proposed structural change I suggest moving this inside utils.